File: example/src/main/scala/com/example/api/sum_columns.scala
package com.example.api

import java.util
import scala.collection.JavaConversions._
import org.apache.spark.sql.{DataFrame, Dataset, Row}
import com.example.SumColumns

object SumColumnsAPI {
  /** A cool function*/

  def sumColumns(df: DataFrame,col1: String,col2: String,newCol: String): DataFrame = {
    SumColumns.sumColumns(df,col1,col2,newCol)
  }
}

File: example/python/methods/sum_columns.py
from pyspark.sql import DataFrame


def sum_columns(df,col1,col2,newCol):
	""" A cool function"""

    # Get the Scala API
    api = df._sc._jvm.com.example.api.SumColumnsAPI

    return DataFrame(api.sumColumns(df._jdf,col1,col2,newCol), df.sql_ctx)